### PR TITLE
Monisha/background subtraction

### PR DESCRIPTION
**Pull request by mg6955**
Background addition module with cloud functionality and config.ini changes

### CODE REVIEW

File: reduction/background_subtraction/background_subtractor.py
Comment (vinayakSharm) [inline reduction/background_subtraction/background_subtractor.py:1]: hi monisha can you remove the background subtraction files from this PR as they will be tested in tamanna's background subtraction PR

File: reconstruction/backgroundaddition/requirements_background_addition.txt
Comment (vinayakSharm) [inline reconstruction/backgroundaddition/requirements_background_addition.txt:1]: hi monisha can you remove the background subtraction files from this PR as they will be tested in tamanna's background subtraction PR

File: reconstruction/backgroundaddition/background_addition.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    encoded_video_name = os.path.join(output_folder, name + '_background.mp4')
+    
+    cmd = f"static_ffmpeg -y -i {output_path} -c:v libx264 -crf 34 -preset veryfast {encoded_video_name}"
+    subprocess.run(cmd, shell=True)
+    
+    os.remove(output_path)
+
+def main():
+    '''
+      """
```
Comment (vinayakSharm) [inline reconstruction/backgroundaddition/background_addition.py:74]: pls fix docstring alignment

File: reconstruction/backgroundaddition/background_addition.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import logging
+import os
+from utilities import ConfigHandler
+from utilities import CloudFunctionality
+import time
+
+
+
+def background_addition(video_path, image_path, output_path):
+    """This methods adds static background to masked image
```
Comment (vinayakSharm) [inline reconstruction/backgroundaddition/background_addition.py:22]: pls fix formatting for docstring

File: reconstruction/backgroundaddition/background_addition.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    Parameters
+    Videopath: path to image
```
Comment (vinayakSharm) [inline reconstruction/backgroundaddition/background_addition.py:24]: docstring variable names do not match the actual variable names

**Review by vinayakSharm — CHANGES_REQUESTED** [review]
looks good. 

few docstring concerns. 

can you auto format with black? some of the spacing is odd. 


only real concern is why the fps of the output video is hardcoded? is this intentional, would you like to have the video output in different fps? :)

File: reduction/background_subtraction/background_subtractor.py
Comment (mg6955) [reply reduction/background_subtraction/background_subtractor.py:1]: Okay!

File: reconstruction/backgroundaddition/background_addition.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    # Get the dimensions of the video frames
+    frame_width = int(cap.get(3))
+    frame_height = int(cap.get(4))
+
+    # Define the codec and create a VideoWriter object to save the output
+    
+    fourcc = cv2.VideoWriter_fourcc(*'mp4v')
+
+    out = cv2.VideoWriter(output_path, fourcc, 30, (frame_width, frame_height))
```
[the remark this answers is not in the document]
Comment (mg6955) [reply reconstruction/backgroundaddition/background_addition.py:43]: I thought the videos we encode has a constant fps of 30s. Should I encode it to the fps of input video?

File: reconstruction/backgroundaddition/background_addition.py (hunk 3 above)
Comment (mg6955) [reply reconstruction/backgroundaddition/background_addition.py:24]: Fixed

File: reconstruction/backgroundaddition/background_addition.py (hunk 2 above)
Comment (mg6955) [reply reconstruction/backgroundaddition/background_addition.py:22]: Fixed

File: reconstruction/backgroundaddition/background_addition.py (hunk 1 above)
Comment (mg6955) [reply reconstruction/backgroundaddition/background_addition.py:74]: Fixed

File: reduction/background_subtraction/requirements_background_subtraction.txt
[the remark this answers is not in the document]
Comment (mg6955) [reply reduction/background_subtraction/requirements_background_subtraction.txt:1]: Removed

File: config.ini [hunk 5]
```diff
@@ -109,6 +109,13 @@ input_prefix_s3 = reduced-videos/${reduction.ffmpeg_resolution_downsampler:metho
 output_prefix_s3 = reconstructed-videos/${method_name}-codec_${codec}/
 model_prefix_s3 = pretrained-models/fsrcnn_x4.pb ; s3 prefix of the pre-trained model
 
+[reconstruction.background_addition]
+; method specific parameters:
+
+; s3 prefixs for input/output:
+method_name = background_addition
+input_prefix_s3 = reduced-videos/background_subtractor/
+output_prefix_s3 = reconstructed-videos/%(method_name)s/
```
Comment (vinayakSharm) [inline config.ini:118]: hi monisha this needs to be a $ sign instead of % sign

**Review by vinayakSharm — APPROVED** [review]

**Pull request merged**